### PR TITLE
Issue #1244, Issue #1043 Use existing field properties and only update as per ALLOW_FIELD_RELAXATION

### PR DESCRIPTION
**Pull request by isha97**

### CODE REVIEW

File: bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryUtil.java [hunk 1]
```diff
@@ -570,15 +575,21 @@ public static Schema adjustSchemaIfNeeded(Schema wantedSchema, Schema existingTa
...
+    Mode fieldMode = existingField.getMode() != null ? existingField.getMode() : NULLABLE;
+    if (allowFieldRelaxation && field.getMode() == NULLABLE) {
+      fieldMode = NULLABLE;
+    }
     if (field.getType().equals(LegacySQLTypeName.NUMERIC)
-        && existingField != null
         && existingField.getType().equals(LegacySQLTypeName.BIGNUMERIC)) {
       // convert type
-      return field.toBuilder().setType(LegacySQLTypeName.BIGNUMERIC).build();
+      return existingField.toBuilder().setMode(fieldMode).build();
```
Comment (vishalkarve15) [inline bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryUtil.java:590]: this if clause is not required

**Comment by vishalkarve15** [comment]
Please edit the title to include Issue #1043 as well

File: bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryUtil.java (hunk 1 above)
Comment (isha97) [reply bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryUtil.java]: Fixed

File: bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryUtil.java [hunk 2]
```diff
@@ -570,15 +575,17 @@ public static Schema adjustSchemaIfNeeded(Schema wantedSchema, Schema existingTa
...
-        && existingField != null
-        && existingField.getType().equals(LegacySQLTypeName.BIGNUMERIC)) {
-      // convert type
-      return field.toBuilder().setType(LegacySQLTypeName.BIGNUMERIC).build();
+  static Field adjustField(
+      Field field, @Nullable Field existingField, boolean allowFieldRelaxation) {
+    if (existingField == null) {
+      return field;
     }
+    Mode fieldMode = existingField.getMode() != null ? existingField.getMode() : NULLABLE;
```
Comment (davidrabinowitz) [inline bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryUtil.java:583]: condition is redundant as you know that `existingField.getMode() != null` based on line 580

**Pull request merged**